pkg/data/helpers.go

```diff
@@ -37,3 +45,52 @@ func getFiltersFromStruct(i interface{}, keyName *string) (*expression.KeyCondit
 	}
 	return kb, cb
 }
+
+func putItem(input interface{}, data interface{}, tableName string, expr *expression.Expression) error {
+
+	returnValue := "NONE"
+	i, _ := input.(*Account)
+	account, _ := data.(model.Account)
+
+	putMap, _ := dynamodbattribute.Marshal(account)
+	err := Invoke(i.DynamoDB, "PutItem",
+		&dynamodb.PutItemInput{
+			// Query in input Table
+			TableName: aws.String(tableName),
+			// Find record for the requested input
+			Item: putMap.M,
+			// Condition Expression
+			ConditionExpression:       expr.Condition(),
+			ExpressionAttributeNames:  expr.Names(),
+			ExpressionAttributeValues: expr.Values(),
+			// Return the updated record
+			ReturnValues: aws.String(returnValue),
+		},
+	)
+	var awsErr awserr.Error
+	if gErrors.As(err, &awsErr) {
+		if awsErr.Code() == "ConditionalCheckFailedException" {
+			return errors.NewConflict(
+				tableName,
+				"input",
+				fmt.Errorf("unable to update %s: Table has been modified since request was made", tableName))
+		}
+	}
+
+	return nil
+}
+
+func Invoke(any interface{}, name string, args... interface{}) error {
+	inputs := make([]reflect.Value, len(args))
+	for i, _ := range args {
+		inputs[i] = reflect.ValueOf(args[i])
+	}
+	result := reflect.ValueOf(any).MethodByName(name).Call(inputs)
+	err := result[0].Interface()
+
+	if err != nil {
```

Comment:
Can you explain this part a little to me?  Seems weird to say err!= nil then return nil for error.

Additional return gErrors.New("error").  Does that mask the original AWS error?  Should we making this an errors.NewInternalServer().

Reply:
This is WIP. Please ignore this for now.